scala-package/core/src/main/scala/org/apache/mxnet/ResourceScope.scala

```diff
@@ -140,12 +134,14 @@ object ResourceScope {
 
     try {
       val ret = body
-       ret match {
+      if (scope == null) {
```

Comment:
why do we need to this only when a scope is not passed? these deallocate resources in a collection when using an existing scope.

Reply:
Removed it before I saw your comment but I'll explain what I was doing anyway.

These keep the return from being deallocated. When explicitly passing in a scope that you want all new native resources to be in, this code removes it from that scope and puts it in whatever happens to be the scope which is 2nd to last. I was considering making that not happen when you pass in a scope (because presumably you want resources to all be at the scope level you specified) but changed my mind.